src/Symfony/Component/Validator/Constraints/AbstractComposite.php

```diff
@@ -0,0 +1,120 @@
+     *
+     * Set of constraints
+     */
+    public $constraints = [];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function __construct($options = null)
+    {
+        parent::__construct($options);
+
+        $this->constraints = (array) $this->constraints;
+
+        // Each constraint contained
+        foreach ($this->constraints as $constraint) {
+            if (!$constraint instanceof Constraint) {
+                throw new ConstraintDefinitionException(sprintf('The value %s is not an instance of Constraint in constraint %s', $constraint, __CLASS__));
+            }
+
+            if ($constraint instanceof Valid) {
+                throw new ConstraintDefinitionException(sprintf('The constraint Valid cannot be nested inside constraint %s. You can only declare the Valid constraint directly on a field or method.', __CLASS__));
+            }
+
+            // If explicit groups are defined
+            if ($this->groups != [self::DEFAULT_GROUP]) {
+                /*
+                 * If constraint has explicit groups defined
+                 *
+                 * In that case, the groups of the nested constraint need to be
+                 * a subset of the groups of the outer constraint.
+                 */
+                if ($constraint->groups !== [self::DEFAULT_GROUP]) {
+                    // If are not a subset
+                    if ($constraint->groups != array_intersect($constraint->groups, $this->groups)) {
+                        throw new ConstraintDefinitionException(sprintf('The groups defined in Constraint %s must be a subset of the groups defined in the Constraint %s', $constraint, __CLASS__));
+                    }
+
+                    // Otherwise, we add all defined groups here
+                } else {
+                    foreach ($this->groups as $group) {
+                        $constraint->addImplicitGroupName($group);
+                    }
+                }
+
+                /*
+                 * Otherwise, we merge current groups with constraint
+                 */
+            } else {
+                $this->groups = array_unique(array_merge($this->groups, $constraint->groups));
+            }
+        }
+    }
+
+    /**
+     * Adds the given group if this constraint is in the Default group.
+     *
+     * Also propagate same method to nested Constraints.
+     *
+     * @param string $group
+     */
+    public function addImplicitGroupName($group)
+    {
+        parent::addImplicitGroupName($group);
+
+        foreach ($this->constraints as $constraint) {
+            $constraint->addImplicitGroupName($group);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefaultOption()
+    {
+        return 'constraints';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getRequiredOptions()
+    {
+        return ['constraints'];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getCompositeOption()
```

Comment:
I do not really see why we need this method.

Reply:
to say if it's a contraints of a field